network-api/networkapi/buyersguide/pagemodels/product_category.py

```diff
@@ -50,8 +49,8 @@ class BuyersGuideProductCategory(models.Model):
 
     @property
     def published_product_count(self):
-        # TODO: REMOVE: LEGACY FUNCTION
-        return Product.objects.filter(product_category=self, draft=False).count()
+        from networkapi.wagtailpages.models import ProductPage
+        return ProductPage.objects.filter(product_categories__category=self).live().count()
```

Comment:
this just duplicates what we already have below, if we're doing that, let's start by making sure that `published_product_count` doesn't get called at all, replaced with `published_product_page_count` instead.

Reply:
Solid point and I couldnt agree more. Will work on this shortly.